src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react"
+import { memo, useCallback, useEffect, useMemo, useState } from 'react';
+import { Person } from "../../types/Person";
+import { peopleFromServer } from "../../data/people";
+import debounce from "lodash.debounce";
+import classNames from "classnames";
+
+interface Props {
+  delay?: number;
+  width?: number;
+  onSelected: (Person: Person | '') => void;
+  onWarning: (a: boolean) => void;
+}
+
+export const AutoComplete: React.FC<Props> = memo(
+  ({ delay = 300, width = 300, onSelected, onWarning }) => {
+    const [query, setQuery] = useState('');
+    const [appliedQuery, setAppliedQuery] = useState('');
+    const [showDrop, setShowDrop] = useState(true);
+    const [isFirstRender, setIsFirstRender] = useState(true);
+
+    const filteredPeople = useMemo(() => {
+      const filter = peopleFromServer.filter(person =>
+        person.name
+          .toLocaleLowerCase()
+          .includes(appliedQuery.toLocaleLowerCase()),
+      );
+
+      return filter;
+    }, [appliedQuery]);
+
+    useEffect(() => {
+      if (!isFirstRender) {
+        if (filteredPeople.length === 0) {
+          onWarning(true);
+          setShowDrop(true);
+        } else {
+          onWarning(false);
+          setShowDrop(false);
+        }
+      } else {
+        setIsFirstRender(false);
+      }
+    }, [filteredPeople]);
+
+    const applyQuery = useCallback(
+      debounce((value: string) => {
+        if (value !== query) {
+          setAppliedQuery(value);
+        }
+      }, delay),
+      [query],
+    );
```

Comment:
`useCallback` is being unable to compute the debounce return's dependencies, `useCallback` expects an inline function as parameter.
Change this to a `useMemo` with an inline function:
```js
const applyQuery = useMemo(() => (
  debounce(
    (value: string) => {
      if (value !== query) {
        setAppliedQuery(value);
      }
    }, 
    delay
  )),
  [query]
);
```